pkg/output/plain.go

```diff
@@ -89,6 +89,33 @@ func (p *PlainPrinter) Print(_ context.Context, r *detectors.ResultWithMetadata)
 
 	for _, data := range meta {
 		for k, v := range data {
+			// Only print Git commit refs when they're associated with a pull/merge request.
```

Comment:
Doing this in the printer doesn't seem like the best approach. Especially as this will need to be copied to other outputs.

Reply:
Since the logic varies between sources (GitHub and GitLab are different), should it go in the individual source metadata funcs? (I.e. only write the commit ref to the source metadata if it's useful.)

Reply:
You'd ideally be able to benefit from this regardless of whether you use `trufflehog git` or `trufflehog github`. The logic is minimal and almost identical between GitHub and GitLab.

Ultimately, this is just finding refs that aren't cloned by default and cloning them. `refs/pull/*` and `/refs/merge-requests/*` happen to have a special meaning for GitHub and GitLab, but other refs could be added in the future which aren't specific to any source.